src/ts/lib/storage.ts

```diff
@@ -108,6 +108,48 @@ class Storage {
 		this.setSpace(obj);
 	};
 
+	setLastOpened (windowId: string, param: any) {
+		const obj = this.get('lastOpened') || {};
+		obj[windowId] = Object.assign(obj[windowId] || {}, param);
+		this.set('lastOpened', obj, true);
+	};
+
+	deleteLastOpenedByObjectId (objectIds: string[]) {
+		const obj = this.get('lastOpened') || {};
+
+		let windowIdsToDelete = Object.keys(obj).reduce(
+			(windowIds, windowId) => !obj[windowId] || objectIds.includes(obj[windowId].id) ? windowIds.concat(windowId) : windowIds, []
+		);
+
+		this.deleteLastOpenedByWindowId(windowIdsToDelete, true);
+	};
+
+	deleteLastOpenedByWindowId (windowIdsToDelete: string[], homeIncluded?: boolean) {
+		if (windowIdsToDelete.length == 0) {
+			return;
+		};
+
+		const obj = this.get('lastOpened') || {};
+
+		if (!homeIncluded) {
+			windowIdsToDelete = windowIdsToDelete.filter((id) => id !== '1');
+		};
+
+		windowIdsToDelete.forEach ((windowId) => delete(obj[windowId]));
+		this.set('lastOpened', obj, true);
+	};
+
+	deleteAllLastOpened () {
+		const obj = this.get('lastOpened') || {};
+		let windowIdsToDelete = Object.keys(obj);
+		this.deleteLastOpenedByWindowId(windowIdsToDelete);
```

Comment:
This is ambiguous,
```
this.set('lastOpened', {}, true);
```

Reply:
`this.set('lastOpened', {}, true);` doesn't have the same effect. It removes all keys. The current call `deleteLastOpenedByWindowId(windowIdsToDelete)` keeps the `'1'` key intact.